kiwi/filesystem/ext3.py

```diff
@@ -24,16 +24,20 @@ class FileSystemExt3(FileSystemBase):
     """
     **Implements creation of ext3 filesystem**
     """
-    def create_on_device(self, label: str = None):
+    def create_on_device(self, label: str = None, kb_size: int = 0):
         """
         Create ext3 filesystem on block device
 
         :param string label: label name
+        :param int kb_size:
+            size in KB, can also be counted from the end via -KB
         """
-        device = self.device_provider.get_device()
+        device = [self.device_provider.get_device()]
         if label:
             self.custom_args['create_options'].append('-L')
             self.custom_args['create_options'].append(label)
+        if kb_size:
+            device.append(self._fs_size(kb_size))
```

Comment:
nit, feel free to ignore it. I am not sure `device` is the best variable name here, that forced me to read and check  `mkfs.ext*` man pages 😅  feels like having multiple devices :)

Reply:
yeah I think part of the problem is that mkfs.ext expects a custom size as a positional parameter. The tooling does not provide an option name or something:

```
mke2fs ... device [ fs-size ]
```

This is how the man page describes it. From the above code it imho becomes clear if one knows how mkfs.extX expects the options.

Do you have an idea how to change it to become a better read ? For me it was so obvious that I'm just blind and can't think of a better alternative :-))

Reply:
I don't know honestly, but probably `device_args` or even simply `args`. My point is that I had to reread `_fs_size` method to check if it was returning some sort of device object, also the fact that `device` is a list did not help. I got confused by reading these lines as I did not notice at a glance it was just collecting a list of command arguments.

Reply:
ok, thanks yes let me change that

Reply:
@davidcassany I pushed a commit to address your issue. I agree it really was not obvious how the device variable was used. Thanks